zproject/backends.py

```diff
@@ -899,6 +900,77 @@ def external_auth_method(cls: Type[ExternalAuthMethod]) -> Type[ExternalAuthMeth
     EXTERNAL_AUTH_METHODS.append(cls)
     return cls
 
+ExternalAuthDataDict = TypedDict('ExternalAuthDataDict', {
```

Comment:
Maybe add a comment explaining that this is a data structure designed to be sent over the network, and thus only has json-friendly fields in it?

Reply:
Added a comment and fixed conflicts. Haven't worked on the other commits yet.